### PR TITLE
Search and Lister Component

### PR DESCRIPTION
**Pull request by virresh**
@review @sakshi-arora1 @rismehta

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes and the overall coverage did not decrease.
- [ ] All unit tests pass on CircleCi.
- [ ] I ran all tests locally and they pass.

### CODE REVIEW

File: ui.apps/src/main/content/jcr_root/apps/core/fd/components/formsportal/searchlister/v1/searchlister/clientlibs/site/css/search.less [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    .cmp-searchlister-search__box {
+        position: relative;
+        width: 35%;
+        display: inline-block;
+
+        // search icon via font awesome.
+        &:before {
+            content: "\f002";
+            position: absolute;
+            font-family: 'Font Awesome 5 Free';
```
Comment (sakshi-arora1) [inline ui.apps/src/main/content/jcr_root/apps/core/fd/components/formsportal/searchlister/v1/searchlister/clientlibs/site/css/search.less:29]: ??, is this font approved?

Comment (virresh) [reply ui.apps/src/main/content/jcr_root/apps/core/fd/components/formsportal/searchlister/v1/searchlister/clientlibs/site/css/search.less:29]: I'm not sure. 
I've taken this from https://github.com/adobe/aem-core-wcm-components/blob/61887d1651b517561caaa85486d3b5304b9d8a2f/examples/ui.apps/src/content/jcr_root/apps/core-components-examples/clientlibs/clientlib-site/styles/components/form-text.less#L33

File: ui.apps/src/main/content/jcr_root/apps/core/fd/components/formsportal/searchlister/v1/searchlister/searchlister.html [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/-->
+
+<sly data-sly-use.model="com.adobe.cq.forms.core.components.models.formsportal.SearchAndLister"
+     data-sly-use.logic="${'searchlister.js' @ layout=model.layout}"
+     data-sly-use.clientlib="${'/libs/granite/sightly/templates/clientlib.html'}"
+     data-sly-use.items="${logic.layoutPath}"></sly>
+<div class="cmp-searchlister" data-cmp-is="formssearch" id="${model.id}" data-queryPath="${resource.path}.model.json">
+    <h1 class="cmp-searchlister__heading" data-sly-text="${model.title}">Forms Portal</h1>
```
Comment (rismehta) [inline ui.apps/src/main/content/jcr_root/apps/core/fd/components/formsportal/searchlister/v1/searchlister/searchlister.html:22]: What about localisation ?

Comment (virresh) [reply ui.apps/src/main/content/jcr_root/apps/core/fd/components/formsportal/searchlister/v1/searchlister/searchlister.html:22]: Localization will be handled at the Sling model level for these strings.
I missed adding it for model title and description (it's there for tooltip text). Will add missing I18n calls.

Comment (rismehta) [reply ui.apps/src/main/content/jcr_root/apps/core/fd/components/formsportal/searchlister/v1/searchlister/searchlister.html:22]: Sling model should be language agnostic, all i18n related code should be part of the mark-up

Comment (virresh) [reply ui.apps/src/main/content/jcr_root/apps/core/fd/components/formsportal/searchlister/v1/searchlister/searchlister.html:22]: okay

Comment (virresh) [reply ui.apps/src/main/content/jcr_root/apps/core/fd/components/formsportal/searchlister/v1/searchlister/searchlister.html:22]: But then these won't be available as localized in the JSON?
Is that fine?

Comment (virresh) [reply ui.apps/src/main/content/jcr_root/apps/core/fd/components/formsportal/searchlister/v1/searchlister/searchlister.html:22]: Localisation cannot be done due to difference of localisation frameworks.
Component will show whatever content was authored as-is.

**Comment by virresh** [comment]
Closing in favor of https://github.com/adobe/aem-core-forms-components/pull/40

**Pull request closed without merging**